docs/guides/dws-on-csm/readme.md

```diff
@@ -0,0 +1,185 @@
+
+### Checkout the DWS repo for Slurm plugins.
+
+```bash
+ncn-m001:~ # git clone https://github.com/DataWorkflowServices/dws-slurm-bb-plugin.git
+```
+
+### Update the Slurm Configuration Template
+
+The changes to the Slurm pod's `/etc/slurm` directory are controlled through a ConfigMap.  These instructions follow section 10.3.2 in [HPE Cray Programming Environment Installation Guide: CSM on HPE Cray EX Systems (22.10) S-8003](https://support.hpe.com/hpesc/public/docDisplay?docLocale=en_US&docId=a00126783en_us) to update and activate the ConfigMap.
+
+Get the slurm-config-templates ConfigMap.
+
+```bash
+ncn-m001:~ # kubectl get configmap -n services slurm-config-templates -o yaml > slurm-config-templates.yaml
+```
+
+Extract the `slurm.conf` file, update it to enable the burst buffer plugin, and write it back into the ConfigMap.
+
+```bash
+ncn-m001:~ # yq r slurm-config-templates.yaml 'data."slurm.conf"' > slurm.conf
+ncn-m001:~ # echo "BurstBufferType=burst_buffer/lua" >> slurm.conf
+ncn-m001:~ # yq w -i slurm-config-templates.yaml 'data."slurm.conf"' "$(cat slurm.conf)"
+```
+
+Add the `burst_buffer.lua` plugin script and `burst_buffer.conf` configuration file to the ConfigMap.
+
+```bash
+ncn-m001:~ # yq w -i slurm-config-templates.yaml 'data."burst_buffer.lua"' "$(cat dws-slurm-bb-plugin/src/burst_buffer/burst_buffer.lua)"
+ncn-m001:~ # yq w -i slurm-config-templates.yaml 'data."burst_buffer.conf"' "$(cat dws-slurm-bb-plugin/src/burst_buffer/burst_buffer.conf)"
+```
+
+Apply the updated ConfigMap resource.
+
+```bash
+ncn-m001:~ # kubectl apply -f slurm-config-templates.yaml
+```
+
+### Restart the Slurm Configuration Job
+
+The Slurm configuration job will process the ConfigMap into a second one that is included in the pod spec.  The following steps to reconfigure Slurm are found in the HPE Cray Programming Installation Guide.
+
+```bash
+ncn-m001:~ # JOBNAME=$(kubectl get job -n services | grep slurm-config | grep -v import | awk '{print $1}')
+ncn-m001:~ # kubectl get job -n services -o yaml $JOBNAME > slurm-config.yaml
+ncn-m001:~ # kubectl delete -f slurm-config.yaml
+ncn-m001:~ # yq d -i slurm-config.yaml spec.template.metadata
+ncn-m001:~ # yq d -i slurm-config.yaml spec.selector
+ncn-m001:~ # kubectl apply -f slurm-config.yaml
+ncn-m001:~ # SLURMCTLD_POD=$(kubectl get pod -n user -lapp=slurmctld -o jsonpath='{.items[0].metadata.name}')
+ncn-m001:~ # kubectl exec -n user $SLURMCTLD_POD -c slurmctld -- scontrol reconfigure
+```
+
+After the configuration job has been restarted, wait about 30 seconds to see the new content appear inside the pod.
+
+If you were only adding or modifying burst_buffer.lua, then Slurm is now ready for your jobs.  If you were also modifying slurm.conf to set the BurstBufferType, or you were adding or modifying burst_buffer.conf, then you must also restart the Slurm pod.
+
+```bash
+ncn-m001:~ # kubectl rollout restart deployment -n user slurmctld
+```
```

Comment:
CSM's slurm deployment also uses a second deployment called `slurmctld-backup`. This deployment also needs to be updated. The safe way to do this would be to:
1) rollout restart the slurmctld-backup deployment
2) `kubectl wait` for the slurmctld-backup deployment to be ready
3) rollout restart the `slurmctld` deployment